src/object/object_primitive.c

```diff
@@ -8280,21 +8280,44 @@ pr_midxkey_compare_element (char *mem1, char *mem2, TP_DOMAIN * dom1, TP_DOMAIN
 
   if ((*(dom1->type->index_readval)) (&buf_val1, &val1, dom1, -1, false, NULL, 0) != NO_ERROR)
     {
-      return DB_UNK;
+      error = true;
+      goto clean_up;
     }
 
   if ((*(dom2->type->index_readval)) (&buf_val2, &val2, dom2, -1, false, NULL, 0) != NO_ERROR)
     {
-      return DB_UNK;
+      error = true;
+      goto clean_up;
     }
 
   c = tp_value_compare_with_error (&val1, &val2, do_coercion, total_order, &comparable);
+
+  goto clean_up;
+
+exit:
   if (!comparable)
     {
       return DB_UNK;
     }
 
   return c;
+
+clean_up:
+  if (!DB_IS_NULL (&val1) && val1.need_clear == true)
```

Comment:
@lucianlazar2211 

It seems it is better to move `clean_up:` before `exit:` and remove `exit:`.